app/src/main/java/com/keylesspalace/tusky/fragment/ViewThreadFragment.java

```diff
@@ -250,32 +254,16 @@ public void onFailure(@NonNull Call<Status> call, @NonNull Throwable t) {
         });
     }
 
-    private void setReblogForStatus(int position, Status status, boolean reblog) {
-        status.setReblogged(reblog);
-
-        if (status.getReblog() != null) {
-            status.getReblog().setReblogged(reblog);
-        }
-
-        StatusViewData.Concrete viewdata = statuses.getPairedItem(position);
-
-        StatusViewData.Builder viewDataBuilder = new StatusViewData.Builder((viewdata));
-        viewDataBuilder.setReblogged(reblog);
-
-        StatusViewData.Concrete newViewData = viewDataBuilder.createStatusViewData();
-
-        statuses.setPairedItem(position, newViewData);
-        adapter.setItem(position, newViewData, true);
-    }
-
     @Override
     public void onFavourite(final boolean favourite, final int position) {
         final Status status = statuses.get(position);
         timelineCases.favouriteWithCallback(statuses.get(position), favourite, new Callback<Status>() {
             @Override
             public void onResponse(@NonNull Call<Status> call, @NonNull Response<Status> response) {
                 if (response.isSuccessful()) {
-                    setFavForStatus(position, status, favourite);
+                    statuses.set(position, response.body());
```

Comment:
There was some reason we had `setFavForStatus()` but I cannot remember now. Did we need to reuse the instance for some reason? Did we try to avoid some visual glitch? Is there a difference in behavior now?

Reply:
Yes there is a difference, now the fav/boost count will update. It works fine, no idea why this was never done.